cdp-agentkit-core/typescript/src/action-providers/cdp/cdpWalletActionProvider.ts

```diff
@@ -1,54 +1,32 @@
-import { ExternalAddress } from "@coinbase/coinbase-sdk";
+import { Coinbase } from "@coinbase/coinbase-sdk";
 import { z } from "zod";
 
 import { CreateAction } from "../actionDecorator";
 import { ActionProvider } from "../actionProvider";
 import { Network } from "../../network";
-import { CdpWalletProvider } from "../../wallet-providers";
+import { CdpWalletProvider, CdpWalletProviderConfig } from "../../wallet-providers";
 
 import { SolidityVersions } from "./constants";
-import {
-  AddressReputationSchema,
-  DeployContractSchema,
-  DeployNftSchema,
-  DeployTokenSchema,
-  RequestFaucetFundsSchema,
-} from "./schemas";
+import { DeployContractSchema, DeployNftSchema, DeployTokenSchema } from "./schemas";
 
 /**
- * CdpActionProvider is an action provider for Cdp.
+ * CdpWalletActionProvider is an action provider for Cdp.
+ *
+ * This provider is used for any action that requires a CDP Wallet.
  */
-export class CdpActionProvider extends ActionProvider<CdpWalletProvider> {
+export class CdpWalletActionProvider extends ActionProvider<CdpWalletProvider> {
   /**
-   * Constructor for the CdpActionProvider class.
-   */
-  constructor() {
-    super("cdp", []);
-  }
-
-  /**
-   * Check the reputation of an address.
+   * Constructor for the CdpWalletActionProvider class.
    *
-   * @param args - The input arguments for the action
-   * @returns A string containing reputation data or error message
+   * @param config - The configuration options for the CdpWalletActionProvider.
    */
-  @CreateAction({
-    name: "address_reputation",
-    description: `
-This tool checks the reputation of an address on a given network. It takes:
+  constructor(config: CdpWalletProviderConfig = {}) {
+    super("cdp", []);
```

Comment:
the names for action providers should be unique. `cdp_wallet` and `cdp_api` respectively?